File: backend/src/main/java/es/udc/fic/corpuslab/modules/project/metrics/ProjectMetricsService.java
package es.udc.fic.corpuslab.modules.project.metrics;

import es.udc.fic.corpuslab.modules.project.metrics.dtos.ProjectMetricsDto;

public interface ProjectMetricsService {

    ProjectMetricsDto getProjectMetrics(String authenticatedEmail, Long projectId);
}

File: frontend/src/modules/project/metrics/services/projectMetricsService.ts
import { api } from '@/app/config/axiosInstance';
import { toProjectMetrics } from '@/modules/project/shared/mappers/projectMappers';
import type { ProjectMetric, ProjectMetricsResponse } from '@/modules/project/shared/types/project';

export async function getProjectMetrics(projectId: number): Promise<ProjectMetric[]> {
  const response = await api.get<ProjectMetricsResponse>(`/projects/${projectId}/metrics`);
  return toProjectMetrics(response.data);
}
